Exceptions raised by PureFluidPhase::phaseOfMatter

**System information**
- Cantera version: current master (237936c74)

**Expected behavior**
The phase of matter should be correctly reported.

**Actual behavior**
An exception is raised for some cases, originating in `Substance::Tsat`.

**To Reproduce**
In Python:
```python
import cantera as ct
co2 = ct.CarbonDioxide()
co2.report()
```

Produces the error:
```
~/src/cantera/interfaces/cython/cantera/thermo.pyx in cantera._cantera.ThermoPhase.report()
    288         >>> print(phase.report())
    289         """
--> 290         return pystr(self.thermo.report(bool(show_thermo), threshold))
    291 
    292     def __call__(self, *args, **kwargs):

CanteraError: 
***********************************************************************
CanteraError thrown by Substance::Tsat:
No convergence
***********************************************************************
```

Similarly:
```python
import cantera as ct
n2 = ct.Nitrogen()
n2.TP = 100, 1000
n2.phase_of_matter
```
Produces an exception which is not caught by Python:
```
terminate called after throwing an instance of 'Cantera::CanteraError'
  what():  
***********************************************************************
CanteraError thrown by Substance::Tsat:
No convergence
***********************************************************************

Aborted (core dumped)
```

So inverting the test results in errors because

```
src/thermo/PureFluidPhase.cpp:96:29: error: 'this' argument to member function 'satPressure' has type 'const Cantera::PureFluidPhase', but function is not marked const
    } else if (pressure() > satPressure(temperature())) {
                            ^~~~~~~~~~~
include/cantera/thermo/PureFluidPhase.h:186:24: note: 'satPressure' declared here
    virtual doublereal satPressure(doublereal t);
                       ^
1 error generated.
```

This seems to be because `phaseOfMatter()` is a `const` function. I'm not sure what all of this means, except that all of the variations I tried of removing `const` and modifying `satPressure()` to account for that change also resulted in errors or `-Woverloaded-virtual`. I wonder if a check of the temperature to make sure it is in the appropriate bounds would be easier?

--- a/interfaces/cython/cantera/_cantera.pxd
+++ b/interfaces/cython/cantera/_cantera.pxd
@@ -140,7 +140,7 @@ cdef extern from "cantera/thermo/ThermoPhase.h" namespace "Cantera":
 
         # miscellaneous
         string type()
-        string phaseOfMatter()
+        string phaseOfMatter() except +translate_exception
         string report(cbool, double) except +translate_exception
         cbool hasPhaseTransition()
         cbool isPure()

--- a/interfaces/cython/cantera/test/test_purefluid.py
+++ b/interfaces/cython/cantera/test/test_purefluid.py
@@ -243,6 +243,14 @@ class TestPureFluid(utilities.CanteraTest):
         self.water.TQ = 300, 0.4
         self.assertEqual(self.water.phase_of_matter, "liquid-gas-mix")
 
+        # These cases work after fixing GH-786
+        n2 = ct.Nitrogen()
+        n2.TP = 100, 1000
+        self.assertEqual(n2.phase_of_matter, "gas")
+
+        co2 = ct.CarbonDioxide()
+        self.assertEqual(co2.phase_of_matter, "gas")
+
 
 # To minimize errors when transcribing tabulated data, the input units here are:
 # T: K, P: MPa, rho: kg/m3, v: m3/kg, (u,h): kJ/kg, s: kJ/kg-K
@@ -296,7 +304,7 @@ class PureFluidTestCases:
 
     def test_has_phase_transition(self):
         self.assertTrue(self.fluid.has_phase_transition)
-    
+
     def test_consistency_temperature(self):
         for state in self.states:
             dT = 2e-5 * state.T

--- a/src/thermo/PureFluidPhase.cpp
+++ b/src/thermo/PureFluidPhase.cpp
@@ -90,7 +90,7 @@ std::string PureFluidPhase::phaseOfMatter() const
         return "supercritical";
     } else if (m_sub->TwoPhase() == 1) {
         return "liquid-gas-mix";
-    } else if (temperature() > satTemperature(pressure())) {
+    } else if (pressure() < m_sub->Ps()) {
         return "gas";
     } else {
         return "liquid";

